Objects/codeobject.c

```diff
@@ -1667,7 +1683,14 @@ code_new_impl(PyTypeObject *type, int argcount, int posonlyargcount,
 static void
 code_dealloc(PyCodeObject *co)
 {
+    assert(Py_REFCNT(co) == 0);
+    Py_SET_REFCNT(co, 1);
     notify_code_watchers(PY_CODE_EVENT_DESTROY, co);
+    if (Py_REFCNT(co) > 1) {
+        Py_DECREF(co);
```

Comment:
Ah right, you mentioned this in debugging, but clearly it didn't stick :) I'll remember now.

It would be nice if _something_ in CI would catch this problem. Even refleak buildbots didn't catch it. I guess it would just hide out until someone manually noticed a discrepancy in `gettotalrefcount()`? If so, it would surprise me if we don't already have such discrepancies hiding some places.